kernel/kernel-api/src/main/java/io/delta/kernel/internal/compaction/LogCompactionWriter.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.compaction;
+
+import static io.delta.kernel.internal.DeltaErrors.wrapEngineExceptionThrowsIO;
+import static io.delta.kernel.internal.lang.ListUtils.getLast;
+import static java.util.Collections.emptyList;
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.data.FilteredColumnarBatch;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.internal.DeltaLogActionUtils;
+import io.delta.kernel.internal.fs.Path;
+import io.delta.kernel.internal.replay.CreateCheckpointIterator;
+import io.delta.kernel.internal.snapshot.LogSegment;
+import io.delta.kernel.internal.util.FileNames.DeltaLogFileType;
+import io.delta.kernel.utils.CloseableIterator;
+import io.delta.kernel.utils.FileStatus;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Utility for writing out log compactions. */
+public class LogCompactionWriter {
+
+  private static final Logger logger = LoggerFactory.getLogger(LogCompactionWriter.class);
+
+  private final Path logPath;
+  private final long startVersion;
+  private final long endVersion;
+  private final long minFileRetentionTimestampMillis;
```

Comment:
Might be worth a short comment explaining _why_ we need this value? The others are very obvious. This has to do with tombstone retention, which I had to go and dig up why.

Reply:
Added